### ISSUE TITLE
[Feature Request]

### ISSUE THREAD

**Issue by maks-rafalko** [jameskokoska/Cashew#469]
**Is your feature request related to a problem? Please describe.**

When you see budget's graph, it only shows where you are on the current day (over or behind expected budget value), but it doesn't show historical progress and doesn't show how much money you are over or behind.

**Describe the solution you'd like**

When I look at the current month's budget graph, I would like to see 2 additional things

1. progress line the shows real spendings compared to expected line
2. and how much money I'm over or behind expected money.

Example: 

<img width="456" alt="image" src="https://github.com/user-attachments/assets/13540f81-6854-496f-9a5c-b7687e9f51f1">

another example

![Screenshot_20241013_193305_Google Play Store](https://github.com/user-attachments/assets/87bdd1a5-2a8e-4218-8829-8e4dcfc08d7f)

Please consider adding such features to the chart.

**Comment by jameskokoska** [comment]
This is already available in Cashew. You can expand or collapse to the current date or to the overall budget period. Please see the FAQ “Reading a Budget Graph”: https://cashewapp.web.app/faq.html#creating-a-budget-guide

**Comment by maks-rafalko** [comment]
Sorry but I don't get why you are saying it's available.

What I see in the app is exactly the same as describet in the docs, so the graph looks like this:

![image](https://github.com/user-attachments/assets/b2d0aeb4-94ce-4bdb-90b4-90df167a8529)

But this is not what this feature request is about.

Let me try to explain again.

By looking into this graph, you don't understand how much money your are over the expected value in the current day. Collapse/expand doesn't add this info as well.

Please look at the first image I've uploaded in the description of this issue - note the label `$606 over` - this is what is missed in the graph.

and then, on the current graph there is no line of expected spendings (slanted line):

<img width="548" alt="image" src="https://github.com/user-attachments/assets/b62c20c9-bdd0-4e42-be4b-a3ecfd676161">

**Comment by maks-rafalko** [comment]
sorry @jameskokoska for bothering you by mention, just want to clarify if we are on the same page

**Comment by jameskokoska** [comment]
As stated in the FAQ that was linked:
'A dashed line is drawn on the current date and the average amount spent per day up to the current day. The intersection indicates where your spending should be at that given time for a budget to stay on track.'

I don't think providing the user with an exact number of the expected value is an important piece of information. This number is dynamic and isn't significant in terms of action to take. Example: If they are 50$ overspent, that doesn't mean they should save 50$, it means they should just stop spending because they are over the amount based on the time past for that period. The 50$ is a meaningless value and will change through time. 

The visual aspects of the UI already indicate to the user if they are overspending in the budget progress bar or the graph. So the important information is already displayed to the user in Cashew.

**Comment by maks-rafalko** [comment]
> If they are 50$ overspent, that doesn't mean they should save 50$, it means they should just stop spending because they are over the amount based on the time past for that period. The 50$ is a meaningless value and will change through time.

hm, I don't get you. If I have $50 overspent, and at the same time I have "You can spend $25 each day for the next 26 days", it exactly means that I have to "save" for 2 days to be on track. This is no doubt useful information.

> The visual aspects of the UI already indicate to the user if they are overspending in the budget progress bar or the graph.

Agree. What I suggest based on the images from competitors, is to improve even more the usefulness of this graph.

Yes, you can see that you are overspending, but you don't understand for how much.

Also, you don't understand what is the history of your "track". Did you overspend the whole 10 days before? Or were you on track the whole period and just started overspending today? This is the second point that is solved by this "history graph".

![image](https://github.com/user-attachments/assets/208a3252-37b9-40b0-bfe2-885d65c69d0f)

I see that we are discussing 2 different problems, but you are answerring only for a badge - one of the problem I raised here. Probably I should have split this issue by 2.
